miri-script/src/arg.rs

```diff
@@ -0,0 +1,112 @@
+use clap::{Parser, Subcommand};
+
+#[derive(Parser, Clone, Debug)]
+#[command(author, about, long_about = None)]
+pub struct Cli {
+    #[command(subcommand)]
+    pub commands: Subcommands,
+}
+
+#[derive(Subcommand, Clone, Debug)]
+pub enum Subcommands {
+    /// Installs the miri driver and cargo-miri.
+    /// Sets up the rpath such that the installed binary should work in any
+    /// working directory. Note that the binaries are placed in the `miri` toolchain
+    /// sysroot, to prevent conflicts with other toolchains.
+    Install {
+        /// Flags that are passed through to `cargo install`.
+        #[arg(trailing_var_arg = true)]
+        flags: Vec<String>,
+    },
+    /// Just build miri.
+    Build {
+        /// Flags that are passed through to `cargo build`.
+        #[arg(trailing_var_arg = true)]
+        flags: Vec<String>,
+    },
+    /// Just check miri.
+    Check {
+        /// Flags that are passed through to `cargo check`.
+        #[arg(trailing_var_arg = true)]
+        flags: Vec<String>,
+    },
+    /// Build miri, set up a sysroot and then run the test suite.
+    Test {
+        /// Flags that are passed through to `cargo test`.
+        #[arg(trailing_var_arg = true)]
+        flags: Vec<String>,
+    },
+    /// Build miri, set up a sysroot and then run the test suite.
+    Bless {
+        /// Flags that are passed through to `cargo test`.
+        #[arg(trailing_var_arg = true)]
+        flags: Vec<String>,
+    },
```

Comment:
FWIW, what we really want is `./miri test --bless`; that just was too annoying to implement with the shell script.

Feel free to just leave a FIXME to do this later.

Reply:
Nah, it's fine - I've thought that `--bless` would be better than having a separate command for it, but I wanted to stay as close to script as possible. If we're fine with such "breaking" changes, that's fine with me.